### PR TITLE
chore: remove react-test-renderer from package.json

### LINKED ISSUE #774: Remove use of react-test-renderer
We currently have 2 test renderers in Psammead and Simorgh - `react-test-renderer` and `@testing-library/react`.

Making use of 2 test renderers will lead to confusion about which one to use and when, especially when they have different ideas about how to write tests.

## Solution
Remove `react-test-renderer` and all use of it. Refactor shallow render tests because `@testing-library/react` does not have or encourage shallow rendering.

---

Some of the benefits of using `@testing-library/react` include:
 - encourages better testing practices and writing more maintainable tests 
 - is designed to enable and encourage writing tests that use components as the end users use them
 - it supports querying for elements in ways that encourage accessibility in components

Find out more about [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro).

This article briefly describes and compares React test renders https://codeburst.io/revisiting-react-testing-in-2019-ee72bb5346f4

Here is more on why [`@testing-library/react` doesn't do shallow rendering](https://kentcdodds.com/blog/why-i-never-use-shallow-rendering).

### PR DESCRIPTION
**Pull request by mykeels**
Resolves #774

**Overall change:** _Remove reference to react-test-renderers._

**Code changes:**

- _Remove react-test-renderer from package.json._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by EinsteinNjoroge — APPROVED** [review]

**Review by rhenshaw56 — CHANGES_REQUESTED** [review]
I think we may need to remove [references to it here](https://github.com/bbc/psammead/blob/latest/packages/utilities/psammead-test-helpers/README.md#exported-functions) as well, maybe reword the readme a bit

**Review by rhenshaw56 — APPROVED** [review]

**Comment by amywalkerdev** [comment]
Branch must update CHANGELOG.md in psammead-test-helpers
Branch must update package-lock.json in psammead-test-helpers
Branch must update package.json in psammead-test-helpers

**Comment by mykeels** [comment]
I'll be closing this PR, because I can't update the branch.  I've opened a [duplicate PR from a fork](https://github.com/bbc/psammead/pull/2423).
